source/grid/tria_description.cc

```diff
@@ -745,6 +745,11 @@ namespace TriangulationDescription
       const Triangulation<dim, spacedim> &              tria,
       const LinearAlgebra::distributed::Vector<double> &partition)
     {
+#ifdef DEAL_II_WITH_MPI
+      if (tria.get_communicator() == MPI_COMM_NULL)
+        AssertDimension(partition.local_size(), 0);
+#endif
+
```

Comment:
Do you want to add a short discussion about the possibilities for communicators to the header file?